ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,668 @@
+  creatingAccount: boolean;
+  errors: any;
+  invalidEmail: boolean;
+  rolesOptions: any;
+  profile: Profile;
+  showAllFieldsRequiredError: boolean;
+  showInstitution: boolean;
+  showAffiliationRole: boolean;
+  showAffiliationOther: boolean;
+  showNext: boolean;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  address: Address;
+  institutionName: string;
+  institutionRole: string;
+  affiliation: string;
+  affiliationRole: string;
+  affiliationOther: string;
+}
+
+const styles = {
+  sectionLabel: {
+    height: '22px',
+    color: colors.primary,
+    fontFamily: 'Montserrat',
+    fontSize: '16px',
+    fontWeight: 600,
+    lineHeight: '22px',
+    paddingBottom: '1.5rem'
+  },
+  section: {
+    width: '12rem'
+  }
+};
+
+const nameLength = 80;
+
+export const Section = (props) => {
+  return <FormSection
+      style={{display: 'flex', flexDirection: 'column', paddingTop: '1rem', ...props.style}}>
+    <label style={styles.sectionLabel}>
+      {props.header}
+    </label>
+    {props.children}
+  </FormSection>;
+};
+
+export class AccountCreation extends React.Component<AccountCreationProps, AccountCreationState> {
+  private usernameCheckTimeout: NodeJS.Timer;
+
+  constructor(props: AccountCreationProps) {
+    super(props);
+    this.state = {
+      errors: undefined,
+      profile: {
+        // Note: We abuse the "username" field here by omitting "@domain.org". After
+        // profile creation, this field is populated with the full email address.
+        username: '',
+        dataAccessLevel: DataAccessLevel.Unregistered,
+        givenName: '',
+        familyName: '',
+        contactEmail: '',
+        currentPosition: '',
+        organization: '',
+        areaOfResearch: '',
+      },
+      usernameCheckInProgress: false,
+      usernameConflictError: false,
+      creatingAccount: false,
+      showAllFieldsRequiredError: false,
+      showInstitution: true,
+      showAffiliationRole: false,
+      showAffiliationOther: false,
+      showNext: false,
+      invalidEmail: false,
+      rolesOptions: [],
+      address: {
+        streetAddress1: '',
+        streetAddress2: '',
+        city: '',
+        state: '',
+        zipcode: '',
+        country: ''
+      },
+      institutionName: '',
+      institutionRole: '',
+      affiliation: '',
+      affiliationRole: '',
+      affiliationOther: ''
+    };
+  }
+
+  // This method will be modified once the story  below will be done
+  // https://precisionmedicineinitiative.atlassian.net/browse/RW-3284
+  createAccount(): void {
+    const {invitationKey, setProfile} = this.props;
+    const profile = this.state.profile;
+    const emailValidRegex = new RegExp(/^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/);
+    this.setState({showAllFieldsRequiredError: false});
+    this.setState({invalidEmail: false});
+    const requiredFields =
+      [profile.givenName, profile.familyName, profile.username, profile.contactEmail,
```

Comment:
Not everybody has two names. There's a classic post about this at https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/.

Likewise, for citizen scientists, they might not have a current position, org, or research area. [this may be a future issue].

Sorry with hitting you with product concerns in a PR.